src/Adapter/Driver/Pdo/Connection.php

```diff
@@ -237,6 +240,9 @@ public function connect()
                     if (isset($charset) && $pdoDriver != 'pgsql') {
                         $dsn[] = "charset={$charset}";
                     }
+                    if (isset($unix_socket)) {
```

Comment:
You *cannot* set the unix socket if a hostname or port is passed, this is really only relevant for mysql as well.